### ISSUE TITLE
Add script to create Formulas for all Google Fonts

### ISSUE THREAD

**Issue by ronaldtse** [fontist/fontist#108]
All open source Google Fonts are available here: https://github.com/google/fonts

This repository is grouped by license type. The three folders with fonts are:
* `apache/*` under the Apache license
* `ofl/*` under the Open Font License
* `ufl/*` under the Ubuntu Font License

Each folder within these license folders contains a font set. e.g.

https://github.com/google/fonts/tree/master/ofl/notosans contains the "Noto Sans" font. It contains these files:
* `METADATA.pb`: the metadata of this font in the ProtoBuffer format.
* `NotoSans-*.ttf` the font files
* `OFL.txt` the license file of this font.

This task is to create a script that can clone the Google Fonts repo, and for every font family, create a Fontist Formula for it that fills in the details.

The `otfinfo` command can be installed on macOS via `brew install lcdf-typetools`. This is the output for `NotoSans-Regular.ttf`:

```
otfinfo --info ../NotoSans-Regular.ttf 
Family:              Noto Sans
Subfamily:           Regular
Full name:           Noto Sans
PostScript name:     NotoSans
Version:             Version 1.04
Unique ID:           Monotype Imaging - Noto Sans
Description:         Designed by Monotype design team
Designer:            Monotype Design team
Designer URL:        http://www.monotypeimaging.com/ProductsServices/TypeDesignerShowcase
Manufacturer:        Monotype Imaging Inc.
Vendor URL:          http://code.google.com/p/noto/
Trademark:           Noto is a trademark of Google Inc. and may be registered in certain jurisdictions.
Copyright:           Copyright 2012 Google Inc. All Rights Reserved.
License URL:         http://www.apache.org/licenses/LICENSE-2.0
License Description: Licensed under the Apache License, Version 2.0
Vendor ID:           MONO
```

From this output, we can create the Fontist Formula and provide these details: Family, Subfamily, Version, Designer, Designer URL, Vendor URL, Copyright, License URL and License Description.

i.e. NotoSans will be represented by this formula: `googlefonts/notosans.rb` (because the repo uses the path `fonts/ofl/notosans`).

```
module Fontist
  module Formulas
    class NotoSansFont < FontFormula

      fullname = "Noto Sans"  # This is from METADATA.pb "name:"
      cleanname = "NotoSans" # This is name with no spaces

      desc fullname
      homepage "http://code.google.com/p/noto/"  # From otfinfo

      resource "#{cleanname}.zip" do
        # NOTE: This URL is common for all Google Fonts
        url "https://fonts.google.com/download?family=Noto%20Sans"
        sha256 "..."
      end

      provides_font(fullname, match_styles_from_file: {
        # NOTE: This information is from METADATA.pb
        # TODO: update this structure after implementation of #109
        "Italic" => "NotoSans-Italic.ttf",
        "Regular" => "NotoSans-Regular.ttf",
        "Bold" => "NotoSans-Bold.ttf",
        "BoldItalic" => "NotoSans-BoldItalic.ttf"
      })

      def extract
        resource("#{cleanname}.zip") do |resource|
          zip_extract(resource) do |fontdir|
            match_fonts(fontdir, fullname)
          end
        end
      end

      def install
        case platform
        when :macos
          install_matched_fonts "$HOME/Library/Fonts/#{cleanname}"
        when :linux
          install_matched_fonts "/usr/share/fonts/truetype/#{cleanname.downcase}"
        end
      end

      test do
        case platform
        when :macos
          assert_predicate "$HOME/Library/Fonts/#{cleanname}/NotoSans-Regular.ttf", :exist?
        when :linux
          assert_predicate "/usr/share/fonts/truetype/#{cleanname.downcase}/NotoSans-Regular.ttf", :exist?
        end
      end

      # This is from METADATA.pb "copyright:"
      copyright "Copyright 2012 Google Inc. All Rights Reserved."

      # From otfinfo "License URL"
      # Notice this font is a MISMATCH -- the font itself uses the Apache 2.0 license,
      # but the Google Font site shows OFL license.
      license_url "http://www.apache.org/licenses/LICENSE-2.0"

      # From GitHub repo folder `fonts/ofl/notosans/OFL.txt
      open_license <<~EOS
        Copyright 2012 Google Inc. All Rights Reserved.

This Font Software is licensed under the SIL Open Font License,
Version 1.1.
...
      EOS
  end
end
```

**Comment by ronaldtse** [comment]
This is necessary for https://github.com/metanorma/metanorma-ogc/issues/174

**Comment by ronaldtse** [comment]
This depends on #109 .

**Comment by ronaldtse** [comment]
Handled by @alexeymorozov, thanks!

**Comment by ronaldtse** [comment]
Most formulas merged, remaining ones detailed in #122

**Comment by ronaldtse** [comment]
Closing now.
